TrainingApp/src/screens/Auth/SignIn.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { Text, TextInput, TouchableHighlight, KeyboardAvoidingView } from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+import ValidationComponent from 'react-native-form-validator';
+import { BaseStyles } from '../../theme';
+import { signInStyles, BackgroundGradientColors } from './style';
+import { AuthWarning } from './AuthWarning';
+import { MAIN_ROUTES, AUTH_ROUTES } from "../../constants/routes";
+
+export class SignIn extends ValidationComponent {
+    constructor(props) {
+        super(props);
+        this.state = { email: '', password: '' };
+        this.emailError = '';
+        this.passwordError = '';
+        this.defaultEmailPlaceholder = 'Email Address';
+        this.defaultPasswordPlaceholder = 'Password';
+    }
+
+    checkEmailField() {
+        this.validateForm();
+
+        if (this.isFieldInError('email')) {
+            this.emailError = this.getErrorsInField('email');
+        } else {
+            this.emailError = '';
+        }
+    }
+
+    checkPasswordField() {
+        this.validateForm();
+
+        if (this.isFieldInError('password')) {
+            this.passwordError = this.getErrorsInField('password');
+        } else {
+            this.passwordError = '';
+        }
+    }
+
+    onChangeEmail(email) {
+        this.setState({ email });
+    }
+
+    onChangePassword(password) {
+        this.setState({ password });
+    }
+
+    onRestorePassword() { }
+
+    validateForm() {
```

Comment:
better to put this in Utils and accept field names as params do these validations there . So in future if we have any other input elements the can use this validation component.